types/src/proof/definition.rs

```diff
@@ -844,17 +824,56 @@ impl TransactionListProof {
             })
             .collect::<Result<Vec<_>>>()?;
 
+        // Verify transaction infos are correct using the ledger info
```

Comment:
we could also just (1) check the transaction hashes match in `TransactionListWithProof::verify`, or (2) take a `transaction_hashes: Option<&[HashValue]>` to avoid the duplication.

I kind of prefer (1). It makes sense that `TransactionInfoListProof` would just prove "these `TransactionInfo`s are guaranteed to be in the chain state" and `TransactionListWithProof` would verify that the list of transactions matches the proven list of `TransactionInfo`s.

Reply:
Yeah, I agree with approach (1) -- and I always found it odd that `TransactionListWithProof` never did the check itself. My only concern with making this change is whether or not we're certain that we someone (e.g., a client in the eco-system or a dependency) isn't doing something odd (e.g., relying on the checks done by each verify() call individually)...